mmv1/products/cloudbuild/api.yaml

```diff
@@ -903,3 +903,89 @@ objects:
 
                         Paths must be absolute and cannot conflict with other volume paths on the same
                         build step or with certain reserved volume paths.
+  - !ruby/object:Api::Resource
+    name: 'GitHubEnterpriseConfig'
+    base_url: projects/{{projectId}}/githubEnterpriseConfigs
+    self_link: projects/{{$projectId}}/githubEnterpriseConfigs/{{configId}}
+    update_verb: :PATCH
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Creating a GitHub Enterprise trigger': 'https://cloud.google.com/build/docs/automating-builds/github/build-repos-from-github-enterprise#api'
+      api: 'https://cloud.google.com/build/docs/api/reference/rest/v1/projects.githubEnterpriseConfigs'
```

Comment:
Unfortunately I do not think we can leverage GCP Cloud Build API programmatically to create github enterprise config. The [GithubEnterpriseConfig](https://cloud.google.com/build/docs/api/reference/rest/v1/projects.githubEnterpriseConfigs) instance contains a set of credentials which can be only exchanged [during Github App handshake process](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app-from-a-manifest), this is not possible programmatically at this moment as explained on this github issue https://github.com/integrations/terraform-provider-github/issues/509

That being said, adding the captioned API resource to Cloud Build Trigger does not make any sense. As of now, Github Enterprise config has to be created manually on Console UI as per the [official doc](https://cloud.google.com/build/docs/automating-builds/github/connect-host-github-enterprise) 

I am happy to see someone else to say otherwise that we could do it programmatically

Reply:
I am working to add another pull request on resource google_cloudbuild_trigger, I believe we simply assume the Github Enterprise Config has been created and reference it in the trigger 

such as 

```
      "github": {
        "push": {
          "branch": "^main$"
        },
        "owner": "myorg",
        "name": "myrepo",
        "enterpriseConfigResourceName": "projects/xxxxxxxx/locations/europe-west2/githubEnterpriseConfigs/xxxxxxxx-f774-445e-be7e-xxxxxx"
      }
```

https://cloud.google.com/build/docs/api/reference/rest/v1/projects.locations.triggers/create